src/wallet/wallet.cpp

```diff
@@ -1167,6 +1170,29 @@ void CWallet::BlockDisconnected(const std::shared_ptr<const CBlock>& pblock) {
 
 
 
+void CWallet::BlockUntilSyncedToCurrentChain() {
+    AssertLockNotHeld(cs_main);
+    AssertLockNotHeld(cs_wallet);
+
+    {
+        // Skip the queue-draining stuff if we know we're caught up with
+        // chainActive.Tip()
+        LOCK(cs_main);
+        const CBlockIndex* initialChainTip = chainActive.Tip();
+
+        if (m_last_block_processed->GetAncestor(initialChainTip->nHeight) == initialChainTip) {
+            return;
+        }
+    }
+
+    std::promise<void> promise;
+    CallFunctionInValidationInterfaceQueue([&promise] {
```

Comment:
I think you've been bitten by GitHub's obnoxious order-commits-by-date-instead-of-git-ordering bug?